connect/transactionRequest.ts

```diff
@@ -0,0 +1,132 @@
+  Authorization,
+  AuthType,
+  cvToValue,
+  MultiSigHashMode,
+  PayloadType,
+  PostCondition,
+  serializeCV,
+  serializePostCondition,
+  SingleSigHashMode,
+  StacksTransaction,
+  VersionedSmartContractPayload,
+} from '@stacks/transactions';
+import { BigNumber } from 'bignumber.js';
+import { createContractCallPromises, generateUnsignedStxTokenTransferTransaction } from '../transactions';
+import { FeesMultipliers, StxPendingTxData } from '../types';
+import { buf2hex } from '../utils/arrayBuffers';
+
+const STX_DECIMALS = 6;
+
+export async function getContractCallPromises(
+  payload: TransactionPayload,
+  stxAddress: string,
+  network: StacksNetwork,
+  stxPublicKey: string,
+  auth: Authorization,
+) {
+  const [unSignedContractCall, contractInterface, coinsMetaData, showPostConditionMessage] =
+    await createContractCallPromises(payload, stxAddress, network, stxPublicKey);
+  if (auth) {
+    unSignedContractCall.auth = auth;
+  }
+  return {
+    unSignedContractCall,
+    contractInterface,
+    coinsMetaData,
+    showPostConditionMessage,
+  };
+}
+
+export async function getTokenTransferRequest(
+  recipient: string,
+  amount: string,
+  memo: string,
+  stxPublicKey: string,
+  feeMultipliers: FeesMultipliers,
+  network: StacksNetwork,
+  stxPendingTransactions: StxPendingTxData,
+  auth?: Authorization,
+) {
+  const unsignedSendStxTx: StacksTransaction = await generateUnsignedStxTokenTransferTransaction(
+    recipient,
+    amount,
+    memo!,
+    stxPendingTransactions?.pendingTransactions ?? [],
+    stxPublicKey,
+    network,
+  );
+  // increasing the fees with multiplication factor
+  const fee: bigint = BigInt(unsignedSendStxTx.auth.spendingCondition.fee.toString()) ?? BigInt(0);
+  if (feeMultipliers?.stxSendTxMultiplier) {
+    unsignedSendStxTx.setFee(fee * BigInt(feeMultipliers.stxSendTxMultiplier));
+  }
+  if (auth) {
+    unsignedSendStxTx.auth = auth;
+  }
+  return unsignedSendStxTx;
+}
+
+export const isMultiSig = (tx: StacksTransaction): boolean => {
+  const hashMode = tx.auth.spendingCondition.hashMode as MultiSigHashMode | SingleSigHashMode;
+  return hashMode === AddressHashMode.SerializeP2SH || hashMode === AddressHashMode.SerializeP2WSH ? true : false;
+};
+
+const cleanMemoString = (memo: string): string => memo.replace('\u0000', '');
+
+function encodePostConditions(postConditions: PostCondition[]) {
+  return postConditions.map((pc) => buf2hex(serializePostCondition(pc)));
+}
+
+export const txPayloadToRequest = (
+  stacksTransaction: StacksTransaction,
+  stxAddress?: string,
+  attachment?: string,
+): TransactionPayload => {
+  const transactionRequest = {
+    attachment,
+    stxAddress,
+    sponsored: stacksTransaction.auth.authType === AuthType.Sponsored,
+    nonce: Number(stacksTransaction.auth.spendingCondition.nonce),
+    fee: Number(stacksTransaction.auth.spendingCondition.fee),
+    postConditions: encodePostConditions(stacksTransaction.postConditions.values as any[]),
```

Comment:
type warning here. do we need to cast to any[]?

you could update the arg types in function encodePostConditions